### DISCUSSION TITLE
Override Options in Select AttachAction

### DISCUSSION THREAD

**Discussion by antoinemy** [filamentphp/filament#5695] [Help]
Hello everyone,

I have two models with the following relationships:

```
USER
|
|- BelongsToMany : Team
```
```
TEAM
|
|- BelongsToMany : User
```

In my Team resource, I have a RelationManager `UserRelationManager.php` with the following table:

```
public static function table(Table $table): Table
{
    return UserResource::table($table)
        ->headerActions([
            Tables\Actions\AttachAction::make()
                ->preloadRecordSelect()
                ->form(fn (AttachAction $action): array => [
                        $action->getRecordSelect()
                            ->allowHtml()
                            ->searchable()
                            ->getSearchResultsUsing(function (string $search) {
                                $users = User::where('name', 'like', "%{$search}%")->limit(50)->get();
                                return $users->mapWithKeys(function ($user) {
                                    return [$user->getKey() => static::getCleanOptionString($user)];
                                })->toArray();
                            })
                            ->getOptionLabelsUsing(function ($value): string {
                                 $user = User::find($value);
                                 return static::getCleanOptionString($user);
                            })
                        ]),
          ]);
}
```

Here I practiced Matthew's trick (https://filamentphp.com/tricks/render-html-in-select-options) to have user avatars in my `Select` and it works when I research. So far so good. Now I would like to initialize it with this same view. And for that, I have to rewrite the `Options` of my `Select` to update my `Preload`:

```
->options(function () {
   // ...
})
```

But here I lose the native implementation of filament with its Relation. That was the reason I was using `$action->getRecordSelect()` to keep the relationship. I can't get the current Model (Team) in my options to filter my Select.

Do you have any tips for:

- Get the current model like this:
```
->options(function (Model $record) {
   // $record doesn't work
   // Need to return Users without Users already present in the current Team
})
```
- Keep the options filtered by passing it my new view
- Other solutions?

By the way, I tried to implement the `->multiple()` function on AttachAction and visually it works. On the other hand, as soon as we save, it saves only the first element. Can't save more than one.

Is this normal?

Thank you all,
Great job, I can't wait to see V3!

**Comment by pxlrbt** [comment]
Sorry, I can't really follow where the actual problem is. Why do you have to overwrite `->options()`? Should the new layout/view be applied through `getOptionLabelsUsing()`?
